mia/commands/definition.py

```diff
@@ -391,12 +391,15 @@ def download_apps():
         if not os.path.isdir(download_path):
             os.makedirs(download_path, mode=0o755)
         apk_path = os.path.join(download_path, apk_info['package_name'])
-        path, http_message = urlretrieve(apk_info['package_url'], apk_path)
+        cache_path = os.path.join(handler.get_workspace_path(), 'resources', 'apps')
+        if not os.path.isdir(cache_path):
+            os.makedirs(cache_path, mode=0o755)
+        path, http_message = urlretrieve(apk_info['package_url'], apk_path, cache_path)
         if any(http_message['status_code'] == code for code in (200, 206)):
             print('   - downloaded %s' %
                   format_file_size(http_message['Content-Length']))
         elif http_message['status_code'] == 416:
-            print('   - already downloaded. Skipping...')
+            print('   - already downloaded. Skipped.')
         else:
             raise
```

Comment:
I just saw that one should use `raise` without a message only in `except` blocks. Please change this to something like:
```
raise Exception('   - error downloading file.')
```